convert2rhel/repo.py

```diff
@@ -47,3 +54,57 @@ def get_rhel_repoids():
     loggerinst.info("RHEL repository IDs to enable: %s" % ", ".join(repos_needed))
 
     return repos_needed
+
+
+def download_repofile(repofile_url):
+    """Download a repository file from a specific URL.
+
+    :raises exceptions.CriticalError: When the repository file URL is inaccessible.
+    :returns str: Contents of the repofile if successfully downloaded.
+    """
+    try:
+        with closing(urllib.request.urlopen(repofile_url, timeout=15)) as response:
+            contents = response.read()
+
+            if not contents:
+                loggerinst.critical_no_exit("")
+                raise exceptions.CriticalError(
+                    id_="REPOSITORY_FILE_EMPTY_CONTENT",
+                    title="No content was availble in requested repository file.",
+                    description="The requested repository file seems to be empty. No content received when checking for url: %s"
+                    % repofile_url,
+                )
+
+            loggerinst.info("Successfully downloaded the requested repofile from %s." % repofile_url)
+            return contents.decode()
+    except urllib.error.URLError as err:
+        raise exceptions.CriticalError(
+            id_="DOWNLOAD_REPOSITORY_FILE_FAILED",
+            title="Failed to download repository file.",
+            description="Failed to download the requested repository file from %s.\n"
+            "Reason: %s" % (repofile_url, err.reason),
+        )
+
+
+def write_temporary_repofile(contents):
+    """Write a temporary repository file inside the :py:TMP_DIR folder.
+
+    :param contents str: The contents to write to the file
+    :returns: The path to the temporary repofile. If failed to write the
+        repofile, it will return None.
+
+    :raises exceptions.CriticalError: In case of not being able to write the
+        repository contents to a file.
+    """
+    repofile_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=TMP_DIR)
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".repo", delete=False, dir=repofile_dir) as f:
+        try:
+            store_content_to_file(filename=f.name, content=contents)
+            return f.name
+        except (OSError, IOError) as err:
+            raise exceptions.CriticalError(
+                id_="WRITE_REPOSITORY_FILE_FAILED",
+                title="Failed to write repository file.",
```

Comment:
```suggestion
                title="Failed to store a repository file",
```
We don't use full stops in titles.